app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailPlayerCrasher.java

```diff
@@ -0,0 +1,159 @@
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.ExoPlaybackException;
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.databinding.ListRadioIconItemBinding;
+import org.schabi.newpipe.databinding.SingleChoiceDialogViewBinding;
+import org.schabi.newpipe.player.Player;
+import org.schabi.newpipe.util.ThemeHelper;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Outsourced logic for crashing the player in the {@link VideoDetailFragment}.
+ */
+public final class VideoDetailPlayerCrasher {
+
+    // This has to be <= 23 chars on devices running Android 7 or lower (API <= 25)
+    // or it fails with an IllegalArgumentException
+    // https://stackoverflow.com/a/54744028
+    private static final String TAG = "VideoDetPlayerCrasher";
+
+    private static final Map<String, Supplier<ExoPlaybackException>> AVAILABLE_EXCEPTION_TYPES =
+            getExceptionTypes();
+
+    private VideoDetailPlayerCrasher() {
+        // No impls
+    }
+
+    private static Map<String, Supplier<ExoPlaybackException>> getExceptionTypes() {
+        final String defaultMsg = "Dummy";
+        final Map<String, Supplier<ExoPlaybackException>> exceptionTypes = new LinkedHashMap<>();
+        exceptionTypes.put(
+                "Source",
+                () -> ExoPlaybackException.createForSource(
+                        new IOException(defaultMsg)
+                )
+        );
+        exceptionTypes.put(
+                "Renderer",
+                () -> ExoPlaybackException.createForRenderer(
+                        new Exception(defaultMsg),
+                        "Dummy renderer",
+                        0,
+                        null,
+                        C.FORMAT_HANDLED
+                )
+        );
+        exceptionTypes.put(
+                "Unexpected",
+                () -> ExoPlaybackException.createForUnexpected(
+                        new RuntimeException(defaultMsg)
+                )
+        );
+        exceptionTypes.put(
+                "Remote",
+                () -> ExoPlaybackException.createForRemote(defaultMsg)
+        );
+
+        return Collections.unmodifiableMap(exceptionTypes);
+    }
+
+    private static Context getThemeWrapperContext(final Context context) {
+        return new ContextThemeWrapper(
+                context,
+                ThemeHelper.isLightThemeSelected(context)
+                        ? R.style.LightTheme
+                        : R.style.DarkTheme);
+    }
+
+    public static void onCrashThePlayer(final Player player, final LayoutInflater layoutInflater) {
+        final Context context = player.getContext();
+        if (!isPlayerAvailable(player)) {
```

Comment:
Why not just check for nullity instead of creating a separate function? Also, `isPlayerAvailable` will always return `true` in this case since if the player was null a NullPointerException would already have been thrown by line above